go.mod

```diff
@@ -1,57 +1,58 @@
 module github.com/openshift/cluster-node-tuning-operator
 
-go 1.22.0
+go 1.23.0
 
-toolchain go1.22.4
+toolchain go1.23.4
```

Comment:
Is go bump is mandatory as part of the k8s bump?

Reply:
Hey Talor thank you for looking!  Good question.  So [you have added](https://github.com/openshift/cluster-node-tuning-operator/commit/4d6cb12e3) the `toolchain` option in NTO 4.17+.  Is it actually needed at all?When removed completely,

```
go mod tidy
go mod vendor
```
doesn't add it back.

When you leave it and do the same thing, it seems to be populated by your current toolchain (in my case old go 1.23.4).  In my view it make no sense to keep the old 1.22.4 as we now require go 1.23 with this PR.  The question is, whether we need the toolchain line at all.

Reply:
now i see that I comment on the wrong line, I was referring to the go bump in general.

if go 1.23 is align with k8s 1.32 then that's super and we're good.

Regarding the toolchain, it was added when I ran:
```
go mod tidy
go mod vendor
```
If now running the same command, removes the toolchain then I'm completely fine to go with `go mod` recommendations